spinn_front_end_common/interface/interface_functions/edge_to_n_keys_mapper.py

```diff
@@ -125,28 +126,29 @@ def _process_application_partition(partition, n_keys_map, graph_mapper):
 
     @staticmethod
     def _process_machine_partition(partition, n_keys_map):
-
-        if not isinstance(partition.pre_vertex,
-                          AbstractProvidesNKeysForPartition):
-            n_keys_map.set_n_keys_for_partition(partition, 1)
-        else:
-            n_keys_map.set_n_keys_for_partition(
-                partition,
-                partition.pre_vertex.get_n_keys_for_partition(partition, None))
-
-        constraints = list()
-        if isinstance(partition.pre_vertex,
-                      AbstractProvidesOutgoingPartitionConstraints):
-            constraints.extend(
-                partition.pre_vertex.get_outgoing_partition_constraints(
-                    partition))
-
-        for edge in partition.edges:
-            if isinstance(edge.post_vertex,
-                          AbstractProvidesIncomingPartitionConstraints):
+        if partition.traffic_type == EdgeTrafficType.MULTICAST:
+            if not isinstance(partition.pre_vertex,
+                              AbstractProvidesNKeysForPartition):
+                n_keys_map.set_n_keys_for_partition(partition, 1)
+            else:
+                n_keys_map.set_n_keys_for_partition(
+                    partition,
+                    partition.pre_vertex.get_n_keys_for_partition(
+                        partition, None))
+
+            constraints = list()
+            if isinstance(partition.pre_vertex,
+                          AbstractProvidesOutgoingPartitionConstraints):
                 constraints.extend(
-                    edge.post_vertex.get_incoming_partition_constraints(
+                    partition.pre_vertex.get_outgoing_partition_constraints(
                         partition))
-        constraints.extend(partition.constraints)
 
-        return constraints
+            for edge in partition.edges:
+                if isinstance(edge.post_vertex,
+                              AbstractProvidesIncomingPartitionConstraints):
+                    constraints.extend(
+                        edge.post_vertex.get_incoming_partition_constraints(
+                            partition))
+            constraints.extend(partition.constraints)
+
+            return constraints
```

Comment:
This code fails to return an object (relying on Python's default) when `partition.traffic_type` is not `MULTICAST`. This feels like suspicious code-smell. If you want to return `None`, write so explicitly.